hydra_provider/src/database_service/hydra_input/messages/sensor/gps_position1.rs

```diff
@@ -0,0 +1,29 @@
+use sqlx::{query, Postgres, Transaction, postgres::PgQueryResult, Error};
+use messages::sensor::GpsPos1;
+use crate::database_service::hydra_input::saveable::SaveableData;
+
+impl SaveableData for GpsPos1 {
+	async fn save(
+		&self,
+		transaction: &mut Transaction<'_, Postgres>,
+		rocket_message_id: i32,
+	) -> Result<PgQueryResult, Error> {
+		query!(
+			"INSERT INTO rocket_sensor_gps_pos_1
+			(rocket_sensor_message_id, time_stamp, status, time_of_week, latitude, longitude, altitude, undulation)
+			VALUES ($1, $2, $3, $4, $5, $6, $7, $8)",
+			rocket_message_id,
+			self.time_stamp as i32,
+			self.status as i32,
+			self.time_of_week as i32,
+			self.latitude as f32,
+			self.longitude as f32,
+			self.altitude as f32,
+			self.undulation as f32,
+	
```

Comment:
Again, I know outside scope of this PR, but you guys might want to reconsider splitting the structs (here for example gps_pos_1 and gps_pos_2) also in the database. The reasons those structs were split in the rocket is because of size constraints when sending via MavLink and CAN-Bus. Those limits don't apply to the database, so it doesn't make really sense to have this design detail carry over to it.

Reply:
How would someone know if a gps_pos_1 message is related to a gps_pos2 message? 

They are indeed split but we can't really assume that they will be sent in sequence.

What if we get 10 gps_pos_1 in sequence? Etc

Reply:
Yeah we should have a session to talk about the DB structure. I was just forwarding the old code here.

Couple of other things that I was thinking about RE the DB:

* I was thinking a NoSQL approach for the time series data would be better so we don't have to deal with table inheritance or having too many tables in a RDB pattern.  There's too many tables that store different bite sized bits of information and I don't think we're ever gonna query those tables individually. Unless I'm mistaken? @seofernando25

* Another thing is as the message/command format changes down the line writing these hardcoded SQL queries are becoming hard to maintain manually and our DB schema is currently written in JS in the `db` folder but the backend is in Rust and usually backend has the responsibility of maintaining the DB schema structure not the front end.

Reply:
> I was thinking a NoSQL approach for the time series data would be better so we don't have to deal with table inheritance or having too many tables in a RDB pattern.

Uhm... I'm not sure how this would work with GraphQL and type safety in general.

> There's too many tables that store different bite sized bits of information 

I would say that is true

>  and I don't think we're ever gonna query those tables individually.

but I wouldn't make this assumption :thinking:  
Anyone could decide they want to see metrics for X,Y points

> Another thing is as the message/command format changes down the line writing these hardcoded SQL queries are becoming hard to maintain manually

If you don't want to use GraphQL you could  instead insert them with hasura which might be easier

> our DB schema is currently written in JS in the db folder but the backend is in Rust and usually backend has the responsibility of maintaining the DB schema structure not the front end.

Yeah thats life :sweat_smile: 

There is stuff like [this](https://docs.rs/turbosql/latest/turbosql/) and [that](https://www.sea-ql.org/blog/)
it would be awesome if we manage to autogenerate schema from the rust types.

Reply:
We can talk about it this weekend.

Reply:
Hum, good point on not knowing if two messages are related. Actually, just realized that we have a similar problem on the rocket and I have some concerns there. I'll talk to Noah about this.

Regarding generating a schema from the Rust, I can help with that. Some macro magic could maybe do the trick. 😁